mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -416,9 +414,6 @@ private boolean updateFlags(MailboxMessage message, long oldModSeq) {
             .orElseThrow(MailboxDeleteDuringUpdateException::new);
         return tryMessageFlagsUpdate(flagUpdateCalculator,
                 mailbox,
-                messageDAO.retrieveMessages(ImmutableList.of(composedMessageIdWithMetaData), FetchType.Metadata, Optional.empty()).join()
-                    .findFirst()
-                    .map(pair -> pair.getLeft().toMailboxMessage(ImmutableList.of()))
-                    .orElseThrow(() -> new MessageDeletedDuringFlagsUpdateException(cassandraId, (CassandraMessageId) messageId)));
+                composedMessageIdWithMetaData);
```

Comment:
It's reloaded on each stage, so it's fresh.

We should also limit READs to the minimum as they are expensive.

So no we should not.

And BTW, it pass concurrency tests, which leads to no problem.